dspace-api/src/main/java/org/dspace/statistics/export/processor/ExportEventProcessor.java

```diff
@@ -0,0 +1,297 @@
+        data.append("&").append(URLEncoder.encode("req_dat", UTF_8)).append("=")
+            .append(URLEncoder.encode(clientUA, UTF_8));
+        data.append("&").append(URLEncoder.encode("rft.artnum", UTF_8)).append("=").
+                append(URLEncoder.encode("oai:" + configurationService.getProperty("dspace.hostname") + ":" + item
+                        .getHandle(), UTF_8));
+        data.append("&").append(URLEncoder.encode("rfr_dat", UTF_8)).append("=")
+            .append(URLEncoder.encode(referer, UTF_8));
+        data.append("&").append(URLEncoder.encode("rfr_id", UTF_8)).append("=")
+            .append(URLEncoder.encode(configurationService.getProperty("dspace.hostname"), UTF_8));
+        data.append("&").append(URLEncoder.encode("url_tim", UTF_8)).append("=")
+            .append(URLEncoder.encode(getCurrentDateString(), UTF_8));
+
+        return data.toString();
+    }
+
+    /**
+     * Get the current date
+     *
+     * @return the current date as a string
+     */
+    protected String getCurrentDateString() {
+        return new DCDate(new Date()).toString();
+    }
+
+    /**
+     * Checks if an item should be processed
+     *
+     * @param item to be checked
+     * @return whether the item should be processed
+     * @throws SQLException
+     */
+    protected boolean shouldProcessItem(Item item) throws SQLException {
+        if (item == null) {
+            return false;
+        }
+        if (!item.isArchived()) {
+            return false;
+        }
+        if (itemService.canEdit(context, item)) {
+            return false;
+        }
+        if (!shouldProcessItemType(item)) {
+            return false;
+        }
+        if (!shouldProcessEntityType(item)) {
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * Checks if the item's entity type should be processed
+     *
+     * @param item to be checked
+     * @return whether the item should be processed
+     * @throws SQLException
+     */
+    protected boolean shouldProcessEntityType(Item item) throws SQLException {
+        Entity entity = entityService.findByItemId(context, item.getID());
+        EntityType type = entityService.getType(context, entity);
+
+        if (type != null && entityTypes.contains(type.getLabel())) {
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Checks if the item should be excluded based on the its type
+     *
+     * @param item to be checked
+     * @return whether the item should be processed
+     */
+    protected boolean shouldProcessItemType(Item item) {
+        if (trackerType != null && trackerValues != null) {
+            List<MetadataValue> types = itemService
+                    .getMetadata(item, trackerType.split("\\.")[0], trackerType.split("\\.")[1],
+                                 trackerType.split("\\.").length == 2 ? null : trackerType.split("\\.")[2], Item.ANY);
```

Comment:
It appears that the value of `trackerType` will look like `some.thing` or `some.thing.else`.  It might be clearer to split it once into a temporary and then just pass the elements of the temporary to `#getMetadata`.  But it still needs comments.

The compiler's dataflow analyzer *may* understand that it can elide the second, third, and fourth identical `split` calls and re-use its own temporary, but why not make this explicit and clear?

Reply:
This should be more clear now in the code